src/v2i-hub/CARMAStreetsPlugin/src/CARMAStreetsPlugin.cpp

```diff
@@ -588,7 +588,16 @@ void CARMAStreetsPlugin::SubscribeSpatKafkaTopic(){
 					auto spat_ptr = std::make_shared<SPAT>();
 					spat_convertor.convertJson2Spat(payload_root, spat_ptr.get());
 					tmx::messages::SpatEncodedMessage spatEncodedMsg;
-					spat_convertor.encodeSpat(spat_ptr, spatEncodedMsg);
+					try
+					{
+						spat_convertor.encodeSpat(spat_ptr, spatEncodedMsg);
+					}
+					catch(...)
```

Comment:
Last thing. Is it possible to catch the specific exception since this will throw a code smell.

Reply:
@dan-du-car  Why did we mark this as resolved. In some of the other methods you catch a TmxException &ex. Can we do this here as well.